packages/schematics/src/app/rules.ts

```diff
@@ -0,0 +1,207 @@
+import { JsonAstObject, JsonParseMode, normalize, parseJsonAst } from '@angular-devkit/core';
+import {
+  apply,
+  branchAndMerge,
+  mergeWith,
+  move,
+  Rule,
+  SchematicContext,
+  SchematicsException,
+  template,
+  Tree,
+  url,
+} from '@angular-devkit/schematics';
+import { NodePackageInstallTask } from '@angular-devkit/schematics/tasks';
+import { getWorkspace, updateWorkspace } from '@schematics/angular/utility/config';
+import {
+  addPackageJsonDependency,
+  NodeDependency,
+  NodeDependencyType,
+} from '@schematics/angular/utility/dependencies';
+import {
+  appendPropertyInAstObject,
+  findPropertyInAstObject,
+  insertPropertyInAstObjectInOrder,
+} from '@schematics/angular/utility/json-utils';
+import { getProject } from '@schematics/angular/utility/project';
+import { getProjectTargets } from '@schematics/angular/utility/project-targets';
+import { of } from 'rxjs';
+import { concatMap, map } from 'rxjs/operators';
+
+import { getLatestNodeVersion, NpmRegistryPackage } from './npmjs-utils';
+import { Schema } from './schema';
+
+export function validateOptions(options: Schema): Rule {
+  return (tree: Tree, context: SchematicContext) => {
+    if (!options.project || options.project === '') {
```

Comment:
One nifty thing is that we don't have to throw an exception here. We can just inspect the `workspace.projects` object. In most cases, there should really only be 1 project there in that object, so you could find the key and then set the project name for them on the options object, thus saving them some effort. This can be done before options is sent to other parts of the code. I don't think it's common to have more than 1 project honestly, and if they have more than 1, there's a good chance the first one is the one they want to instrument with augury.

IDEs should also pass in the project name. I know WebStorm does.

Reply:
That's one thought I actually had as well, although I wasn't sure we wanted to just assume or install it to a project unless they specifically said yes to it. That's why I wanted to have the `Want to use DEFAULT_PROJECT (Y/n)?` to make sure we have the right project.
I do however agree with you about that probably the most common case there is only one project in the workspace.

I tried using the `"$source": "projectName"` as default value in the schematics, but then the prompt won't display. 

Thoughts? Do you think it's okay to default it to the first `workspace.projects` in case there is none entered?

Reply:
@feeloor I personally would default to workspace.projects. I think that's the intent of 99% of users honestly. No reason to bug them for information we can get automatically. If they don't like the default behaviour, they can always provide the project value manually. Many throw an error if 2 or more projects exist just to be safe, but even that might not be necessary.

Reply:
@egervari - Yeah after looking at how `@angular/material` and other libraries does it I removed the prompt and use the `defaultProject`.